test/browser/cloneElement.test.js

```diff
@@ -84,4 +84,10 @@ describe('cloneElement', () => {
 		const clone = cloneElement(div, { key: 'myKey' });
 		expect(clone.props.key).to.equal(undefined);
 	});
+
+	it('should prevent undefined properties from overriding default props', () => {
+		const element = <div style={{ color: 'blue' }}>thing</div>;
+		const clone = cloneElement(element, { color: undefined });
```

Comment:
Let's update this test to use a Component that defines `defaultProps` to exercise the new code path. Perhaps something like:
```js
class Example extends Component {
  render(props) {
    return <div style={{ color: props.color }}>thing</div>;
  }
}

Example.defaultProps = { color: blue };

const element = <Example color="red" />
const clone = cloneElement(Example, { color: undefined });
```

or somethin like that

Reply:
this is a good point! Adjusted, thank you